pkg/reconciler/configmap/configmap.go

```diff
@@ -0,0 +1,200 @@
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	//TODO eventually we'll need to decide if we want to make this tuneable
+	contextTimeout      = 300 * time.Second
+	ConfigMapName       = "jvm-build-config"
+	CacheDeploymentName = "jvm-artifact-cache-new"
+	EnableRebuilds      = "enable-rebuilds"
+	CacheImage          = "cache-image"
+)
+
+var (
+	log = ctrl.Log.WithName("configmap")
+)
+
+type ReconcileArtifactBuild struct {
+	client        client.Client
+	scheme        *runtime.Scheme
+	eventRecorder record.EventRecorder
+}
+
+func newReconciler(mgr ctrl.Manager) reconcile.Reconciler {
+	return &ReconcileArtifactBuild{
+		client:        mgr.GetClient(),
+		scheme:        mgr.GetScheme(),
+		eventRecorder: mgr.GetEventRecorderFor("ArtifactBuild"),
+	}
+}
+
+func (r *ReconcileArtifactBuild) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	// Set the ctx to be Background, as the top-level context for incoming requests.
+	ctx, cancel := context.WithTimeout(ctx, contextTimeout)
+	defer cancel()
+	if request.Name != ConfigMapName {
+		return reconcile.Result{}, nil
+	}
+	configMap := corev1.ConfigMap{}
+	err := r.client.Get(ctx, request.NamespacedName, &configMap)
+	if err != nil {
+		return reconcile.Result{}, nil
+	}
+	enabled := configMap.Data[EnableRebuilds] == "true"
+	if enabled {
+		r.setupRebuilts(ctx, request)
+	}
+
+	if err = r.setupCache(ctx, request, enabled, configMap); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileArtifactBuild) setupCache(ctx context.Context, request reconcile.Request, rebuildsEnabled bool, configMap corev1.ConfigMap) error {
+	cache := v1.Deployment{}
+	deploymentName := types.NamespacedName{Namespace: request.Namespace, Name: CacheDeploymentName}
+	err := r.client.Get(ctx, deploymentName, &cache)
+	create := false
+	if err != nil {
+		if errors.IsNotFound(err) {
+			create = true
+			cache.Name = deploymentName.Name
+			cache.Namespace = deploymentName.Namespace
+			var replicas int32
+			replicas = 1
+			cache.Spec.Replicas = &replicas
+			cache.Spec.Selector = &v12.LabelSelector{MatchLabels: map[string]string{"app": CacheDeploymentName}}
+			cache.Spec.Template.ObjectMeta.Labels = map[string]string{"app": CacheDeploymentName}
+			cache.Spec.Template.Spec.Containers = []corev1.Container{{
+				Name:            "hacbs-jvm-cache",
+				ImagePullPolicy: "Always",
+				SecurityContext: &corev1.SecurityContext{RunAsUser: i64a(0)},
+				Ports: []corev1.ContainerPort{{
+					Name:          "http",
+					ContainerPort: 8080,
+					Protocol:      "TCP",
+				}},
+				Resources: corev1.ResourceRequirements{
+					Requests: map[corev1.ResourceName]resource.Quantity{"memory": resource.MustParse("128Mi"), "cpu": resource.MustParse("250m")},
+					Limits:   map[corev1.ResourceName]resource.Quantity{"memory": resource.MustParse("700Mi"), "cpu": resource.MustParse("500m")},
```

Comment:
We will need some form of configuration here. Big workspaces may need caches with more resources.

Reply:
yep .... and since my comment we had the cabal discussion with @sbose78 and the rest of the 3 pipeline service teams and reached agreement that there is a broader concern around setting up workspaces/namespaces with more quota based on request and additional payment, not just jvm build service

that said, in the interim, I think you should define keys in the configmap for these, where the values you've picked now could be the default if those key are missing.

Reply:
We definitly want it to be system level configurable, probably also individually configurable (assuming we have some kind of per namespace quota, so a user can't just increment this to get more total memory).

Reply:
we can assume per namespace quota

and I'm good with staging system level vs. individual configuration separately (with system first) if that seems like the correct path to you @stuartwdouglas 

do you want to include  in this PR or a follow up?